muikku-atests/src/test/java/fi/muikku/mock/PyramusMock.java

```diff
@@ -0,0 +1,634 @@
+import fi.muikku.mock.model.MockStudent;
+import fi.pyramus.rest.model.ContactType;
+import fi.pyramus.rest.model.CourseStaffMember;
+import fi.pyramus.rest.model.CourseStaffMemberRole;
+import fi.pyramus.rest.model.CourseStudent;
+import fi.pyramus.rest.model.CourseType;
+import fi.pyramus.rest.model.EducationType;
+import fi.pyramus.rest.model.EducationalTimeUnit;
+import fi.pyramus.rest.model.Email;
+import fi.pyramus.rest.model.Grade;
+import fi.pyramus.rest.model.GradingScale;
+import fi.pyramus.rest.model.Person;
+import fi.pyramus.rest.model.StaffMember;
+import fi.pyramus.rest.model.Student;
+import fi.pyramus.rest.model.StudyProgramme;
+import fi.pyramus.rest.model.StudyProgrammeCategory;
+import fi.pyramus.rest.model.Subject;
+import fi.pyramus.rest.model.WhoAmI;
+import fi.pyramus.webhooks.WebhookCourseStaffMemberCreatePayload;
+import fi.pyramus.webhooks.WebhookCourseStudentCreatePayload;
+import fi.pyramus.webhooks.WebhookPersonCreatePayload;
+import fi.pyramus.webhooks.WebhookStaffMemberCreatePayload;
+import fi.pyramus.webhooks.WebhookStudentCreatePayload;
+
+public class PyramusMock {
+  
+  private List<MockStudent> students = new ArrayList<>();
+  private List<MockStaffMember> staffMembers = new ArrayList<>();
+  private List<Person> persons = new ArrayList<>();
+  private ObjectMapper objectMapper;
+  private HashMap<GradingScale, List<Grade>> gradingScales = new HashMap<>();
+  private List<EducationalTimeUnit> educationalTimeUnits = new ArrayList<>();
+  private List<EducationType> educationTypes = new ArrayList<>();
+  private List<Subject> subjects = new ArrayList<>();
+  private List<StudyProgrammeCategory> studyProgrammeCategories = new ArrayList<>();
+  private List<StudyProgramme> studyProgrammes = new ArrayList<>();
+  private List<CourseType> courseTypes = new ArrayList<>();
+  private HashMap<Long, List<CourseStudent>> courseStudents = new HashMap<>();
+  private HashMap<Long, List<CourseStaffMember>> courseStaffMembers = new HashMap<>();
+  private List<String> payloads = new ArrayList<>();
+  
+  private PyramusMock() {
+      // Prevent direct use
+  }
+
+  public static Builder mocker() {
+      return new PyramusMock.Builder();
+  }
+
+  public static class Builder {
+      private PyramusMock pmock = new PyramusMock();
+
+      public Builder() {
+//        Some defaults for mocks
+        pmock.objectMapper = new ObjectMapper().registerModule(new JodaModule()).disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+        
+        GradingScale gs = new GradingScale(1l, "Pass/Fail", "Passed or failed scale", false);
+        List<Grade> grades = new ArrayList<>();
+        grades.add(new Grade(1l, "Excellent", "Excellent answer showing expertise in area of study", 1l, true, "0", null, false));
+        grades.add(new Grade(2l, "Failed", "Failed answer. Not proving any expertise in the matter.", 1l, false, "1", null, false));
+        pmock.gradingScales.put(gs, grades);
```

Comment:
This code would be much easier to read and less error prone if access to these variables would be encapsulated into methods (e.g. pyramusMock.addGrade(gradingScale, grade))